### ISSUE TITLE
Cannot login sometimes, only works after manually deleting session file

### ISSUE THREAD

**Issue by phanxgames** [phanxgames/AetherStory#20]
The session file appears to sometimes self-corrupt.  Needs to self recover.

**Comment by CharlesIV** [comment]
Manually editing session.db results:
1. Deleting several characters or replacing characters with different ones causes login information to be forgot and the session.db is recreated
2. Editing the file and replacing a character with itself, such as a '-' with a '-', the login fails with: "Login fail...". Further editing of the file or deleting of it fixes this.

**Comment by phanxgames** [comment]
Cannot reproduce. Need instructions. 
Please send me a corrupt session.db that produces the "Login fail..." error.
Thank you.

**Comment by CharlesIV** [comment]
1. Delete something, like a small section, within the file.
2. Login and have the game fix the file.
3. Replace characters with themselves throughout the file and relaunch the game.
4. Repeat step 3 till it fails.

**Comment by phanxgames** [comment]
`
DATABASE settings.db opened successfully!
SQLError: 'Error #3115', details:'no such table: 'settings'', operation:'execute', detailID:'2013'

DATABASE session.db opened successfully!
<success ... />

SQLError: 'Error #3123', details:'', operation:'execute'
FAILURE Login failed. 
`

**Comment by phanxgames** [comment]
Could you please send or attach me a corrupt session.db for testing. Thank you.

**Comment by AthenaOfDelphi** [comment]
[session (2).zip](https://github.com/phanxgames/AetherStory/files/4514307/session.2.zip)

I've just restored this from the recycle bin.

Yesterday morning, I lost the ability to login... kept getting login failed.  This coincided with me signing up another account (to be my player account - current on is named AthenaOfDelphi which is better for my GM account so there is no confusion).

I think it started when I tried to login with the new account, which of course doesn't have a product key attached to it, but I can't be 100% sure on that.

Essentially I tried numerous times to login.  I eventually gave up and quit trying.

Later in the day, when Josh was having a problem I tried to login, it failed and so I deleted the file I've attached to this.  At that point it was remade and I was able to login normally.

I'm suspecting this may be an SQLite issue relating to the content of the file not being properly updated before the file is closed.  It's been a while since I did anything with SQLite, but IIRC if you don't close the DB properly (and thus give it a chance to properly flush) it can result in corruption.

https://www.sqlite.org/howtocorrupt.html
